### ISSUE TITLE
Write unused function check

### ISSUE THREAD

**Issue by jonsyu1** [palantir/checks#10]
I believe `golint` or `govet` already catches unused variables, but we should check for unused functions as well.
Example of unused functions: https://github.palantir.build/deployability/skylab/pull/4570

**Comment by 14rcole** [comment]
I agree that this would be a valuable addition, but do you see this working with public functions or only private functions?  If you see a use case for public functions as well, do you see it checking only the package in which the function exists or the entire project?  Obviously external projects can't necessarily be checked, but this might give a sense of which functions don't need to be public.

**Comment by jonsyu1** [comment]
Only private functions, just like the go linter's unused variable check.
